### PR TITLE
Update our build with latest .NET Core 3.0 SDK

### PR DESCRIPTION
**Pull request by weshaggard**

### CODE REVIEW

**Comment by pakrym** [comment]
:shipit:

**Comment by weshaggard** [comment]
@AlexGhiondea do you have any issues with us moving to the .NET Core 3 SDK? 

@pakrym I'm doing some scouting but I'm not planning to enable it just yet. We are refactoring our build pipelines so I want that to land first and I also need to make sure we have a pointer to what version of VS is needed to work with this SDK as well.

**Comment by AlexGhiondea** [comment]
Nope! Curious if all goes well :).

**Comment by weshaggard** [comment]
Fixes https://github.com/Azure/azure-sdk-for-net/issues/5752

**Comment by weshaggard** [comment]
@mitchdenny would you mind having a quick review of these pipeline changes?

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ trigger:
   - master
 
 variables:
-  DotNetCoreVersion: '2.1.503'
+  DotNetCoreSDKVersion: '3.0.100-preview5-011568'
```
Comment (mitchdenny) [inline .azure-pipelines/client.yml]: Should we take this opportunity for this template to reference the globals template?

File: eng/pipelines/templates/variables/globals.yml [hunk 2]
```diff
@@ -1,2 +1,3 @@
 variables:
-  DotNetCoreVersion: '2.1.503'
\ No newline at end of file
+  DotNetCoreSDKVersion: '3.0.100-preview5-011568'
+  DotNetCoreRuntimeVersion: '2.1.10'
```
Comment (mitchdenny) [inline eng/pipelines/templates/variables/globals.yml]: Is going back to 2.1.10 intentional here - we were already on 2.1.503?

File: .azure-pipelines/client.yml (hunk 1 above)
Comment (weshaggard) [reply .azure-pipelines/client.yml]: I didn't think it was needed given this yml file will go away soon.

File: eng/pipelines/templates/variables/globals.yml (hunk 2 above)
Comment (mitchdenny) [reply eng/pipelines/templates/variables/globals.yml]: Oh SDK vs. Runtime - I think I see what you are doing.

**Review by mitchdenny — APPROVED** [review]
LGTM, my only comment (optional) is that you consider referencing the global variables template in client.yml.

**Pull request merged**